link.h not found on mac os

It looks like link.h is not available on mac os:

```
/Users/Jule/Documents/PhD/aspect/source/main.cc:30:12: fatal error:
'link.h'
 file not found
# include <link.h>
```

It is probably enough to remove this shared_lib validation on mac.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -134,6 +134,19 @@ ELSE()
   ENDFOREACH()
 ENDIF()
 
+# See whether we can verify that every plugin we load is compiled against
+# the same deal.II library
+INCLUDE (CheckIncludeFiles)
+CHECK_INCLUDE_FILES ("link.h" _HAVE_LINK_H)
+IF (_HAVE_LINK_H)
+  MESSAGE(STATUS "Enabling checking of compatible deal.II library when loading plugins")
+  FOREACH(_source_file ${TARGET_SRC})
+    SET_PROPERTY(SOURCE ${_source_file}
+      APPEND PROPERTY COMPILE_DEFINITIONS ASPECT_HAVE_LINK_H=1)
+  ENDFOREACH()
+ENDIF()
+
+
 DEAL_II_INVOKE_AUTOPILOT()
 
 MESSAGE(STATUS "Writing config into detailed.log...")

--- a/source/main.cc
+++ b/source/main.cc
@@ -27,7 +27,9 @@
 
 #if ASPECT_USE_SHARED_LIBS==1
 #  include <dlfcn.h>
-#  include <link.h>
+#  ifdef ASPECT_HAVE_LINK_H
+#    include <link.h>
+#  endif
 #endif
 
 
@@ -124,6 +126,7 @@ int get_names_of_shared_libs (struct dl_phdr_info *info,
 // has deal.II only once
 void validate_shared_lib_list (const bool before_loading_shared_libs)
 {
+#ifdef ASPECT_HAVE_LINK_H
   // get the list of all shared libs we currently link against
   std::set<std::string> shared_lib_names;
   dl_iterate_phdr(get_names_of_shared_libs, &shared_lib_names);
@@ -170,6 +173,10 @@ void validate_shared_lib_list (const bool before_loading_shared_libs)
       else
         throw aspect::QuietException();
     }
+#else
+  // simply mark the argument as read, to avoid compiler warnings
+  (void)before_loading_shared_libs;
+#endif
 }
 
 
@@ -237,8 +244,8 @@ void possibly_load_shared_libs (const std::string &parameter_filename)
           std::cerr << std::endl << std::endl
                     << "----------------------------------------------------"
                     << std::endl;
-          std::cerr << "You can not load additional shared libraries on " << std::endl
-                    << "systems where you link ASPECT as a static executable."
+          std::cerr << "You can not load plugins through additional shared libraries " << std::endl
+                    << "on systems where you link ASPECT as a static executable."
                     << std::endl
                     << "----------------------------------------------------"
                     << std::endl;

